src/main/java/com/bathanh/apibook/domain/user/UserService.java

```diff
@@ -0,0 +1,60 @@
+package com.bathanh.apibook.domain.user;
+
+import com.bathanh.apibook.error.UserAvailableException;
+import com.bathanh.apibook.persistence.user.UserStore;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static com.bathanh.apibook.domain.user.UserError.supplyUserNotFound;
+
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserStore userStore;
+
+    public List<User> findAllUsers() {
+        return userStore.findAllUsers();
+    }
+
+    public User findUserById(final UUID id) {
+        return userStore.findUserById(id)
+                .orElseThrow(supplyUserNotFound(id));
+    }
+
+
+    private void verifyUserAvailable(final User user) {
+        final Optional<User> userOptional = userStore.findUserByUserName(user.getUsername());
+        if (userOptional.isPresent()) {
+            throw new UserAvailableException("User Available!");
```

Comment:
```suggestion
            throw new UserAvailableException("User Available!");
```

Message không phù hợp

Reply:
resolved